linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java

```diff
@@ -61,12 +64,36 @@ private void run(String bomCoordinates)
     Bom baseline = RepositoryUtility.readBom(bomCoordinates);
     ImmutableSet<SymbolProblem> problemsInBaseline =
         LinkageChecker.create(baseline).findSymbolProblems().keySet();
-
     Bom snapshot = copyWithSnapshot(repositorySystem, baseline);
 
+    // Compare coordinates of the two BOM. No need to run comparison if they are the same.
+    ImmutableList<String> baselineCoordinates =
+        baseline.getManagedDependencies().stream()
+            .map(Artifacts::toCoordinates) // DefaultArtifact does not override equals
+            .collect(toImmutableList());
+    ImmutableList<String> snapshotCoordinates =
+        snapshot.getManagedDependencies().stream()
+            .map(Artifacts::toCoordinates)
+            .collect(toImmutableList());
+    if (baselineCoordinates.equals(snapshotCoordinates)) {
+      System.out.println("No snapshot version for the BOM found. Not running comparison.");
```

Comment:
This doesn't feel quite accurate. The snapshot was found, but the answer is the same as below. This is just an optimization.

Reply:
Updated. Does this look good now?